src/sections/trade/sections/otc/orders/OtcOrdersData.utils.ts

```diff
@@ -1,42 +1,109 @@
 import { useMemo } from "react"
 import { useOrdersData, useOrdersState, getOrderStateValue } from "api/otc"
 import BN from "bignumber.js"
+import { useAssetPrices } from "utils/displayAsset"
 
 export const useOrdersTableData = () => {
   const treasuryAddr = import.meta.env.VITE_TRSRY_ADDR
   const orders = useOrdersData()
   const orderIds = orders.data?.map((order) => order.id)
   const ordersState = useOrdersState(orderIds || [])
 
-  const queries = [orders, ...ordersState]
+  const assets = orders.data
+    ?.flatMap((order) => [
+      {
+        id: order.assetIn?.id,
+        name: order.assetIn?.name,
+        symbol: order.assetIn?.symbol,
+      },
+      {
+        id: order.assetOut?.id,
+        name: order.assetOut?.name,
+        symbol: order.assetOut?.symbol,
+      },
+    ])
+    .filter(
```

Comment:
could you include `filter` fn inside `reduce` to reduce a number of iterations?

Reply:
Resolved in next push.